packages/loader/container-loader/src/container.ts

```diff
@@ -1777,7 +1779,10 @@ export class Container
 	private get client(): IClient {
 		const client: IClient =
 			this.options?.client !== undefined
-				? (this.options.client as IClient)
+				? // this.options.client is shared by ALL containers spawned by the same Loader (including the summarizer)
+				  // We're doing a cheap clone here to ensure it doesn't get tainted by the merge() that happens below, since
+				  // the override should only apply to the individual container it is given to.
+				  (JSON.parse(JSON.stringify(this.options.client)) as IClient)
```

Comment:
I just don't like that this runs on every call to the getter, but that's probably still not frequent enough for this to be a huge perf concern.

Reply:
Yep, also hoping we nuke this thing entirely soon.